File: cliente/src/app/modulos/comparaciones/modelos-comparaciones.ts
export type TipoDiferencia = 'Agregada' | 'Eliminada' | 'Modificada';

export interface VersionComparadaResumen {
  id: number;
  numero: number;
  hash: string;
  capturadaEn: string;
  estable: boolean;
}

export interface ComparacionResumen {
  id: number;
  dispositivoId: number;
  dispositivo: string;
  versionOrigen: VersionComparadaResumen;
  versionDestino: VersionComparadaResumen;
  usuario: string;
  fecha: string;
  estado: 'Pendiente' | 'EnProceso' | 'Completado' | 'Fallido';
  agregadas: number;
  eliminadas: number;
  modificadas: number;
}

export interface DiferenciaComparacionResumen {
  linea: number;
  tipo: TipoDiferencia;
  textoAnterior: string | null;
  textoNuevo: string | null;
}

export interface ComparacionDetalle extends ComparacionResumen {
  diferencias: DiferenciaComparacionResumen[];
}

File: servidor/fuente/RedDiff.Aplicacion/Comparaciones/ModelosComparaciones.cs
namespace RedDiff.Aplicacion.Comparaciones;

public sealed record CompararVersionesSolicitud(
    long VersionOrigenId,
    long VersionDestinoId);

public sealed record VersionComparadaResumen(
    long Id,
    int Numero,
    string Hash,
    DateTimeOffset CapturadaEn,
    bool Estable);

public sealed record ComparacionResumen(
    long Id,
    long DispositivoId,
    string Dispositivo,
    VersionComparadaResumen VersionOrigen,
    VersionComparadaResumen VersionDestino,
    string Usuario,
    DateTimeOffset Fecha,
    string Estado,
    int Agregadas,
    int Eliminadas,
    int Modificadas);

public sealed record DiferenciaComparacionResumen(
    int Linea,
    string Tipo,
    string? TextoAnterior,
    string? TextoNuevo);

public sealed record ComparacionDetalle(
    long Id,
    long DispositivoId,
    string Dispositivo,
    VersionComparadaResumen VersionOrigen,
    VersionComparadaResumen VersionDestino,
    string Usuario,
    DateTimeOffset Fecha,
    string Estado,
    int Agregadas,
    int Eliminadas,
    int Modificadas,
    IReadOnlyList<DiferenciaComparacionResumen> Diferencias);

public sealed record DiferenciaCalculada(
    int Linea,
    string? TextoAnterior,
    string? TextoNuevo);
